django/core/management/commands/shell.py

```diff
@@ -114,6 +116,11 @@ def handle(self, **options):
             exec(options['command'])
             return
 
+        if select.select([sys.stdin], [], [], 0.0)[0]:
```

Comment:
The test passing seems to indicate that `select()` will ignore file handlers on windows, and basically the interpreter will never go inside that if block on windows.

So, it doesn't seem to break when being ran on windows - I'm really not a windows expert - but it's never going to work, seems like on windows, select is just going to ignore the stdin file handler and thus not return that it has data for reading.